File: 4기-코딩테스트-220608/4기 코딩테스트_220608/4기 - 1st 코딩테스트/SmallestSameIndexWithSorted/SmallestSameIndexWithSorted_test.js
import assert from 'assert'
import solution from './SmallestSameIndexWithSorted.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const nums = [1, 3, 5, 3, 2]
    const n = 3

    // when
    const result = solution(nums, n)

    // then
    assert.deepStrictEqual(2, result)
    done();
  });

  it('test 2', function (done) {
    // given
    const nums = [2, 4, 2, 4, 2, 4, 4]
    const n = 4

    // when
    const result = solution(nums, n)

    // then
    assert.deepStrictEqual(3, result)
    done();
  });

  it('test 3', function (done) {
    // given
    const nums = [7, 7, 7, 6, 5, 10, 4, 4, 1, 8]
    const n = 6

    // when
    const result = solution(nums, n)

    // then
    assert.deepStrictEqual(4, result)
    done();
  });

  it('test 4', function (done) {
    // given
    const nums = []
    const n = 100

    // when
    const result = solution(nums, n)

    // then
    assert.deepStrictEqual(-1, result)
    done();
  });

  it('test 5', function (done) {
    // given
    const nums = [1, 3, 5, 3, 2]
    const n = 99

    // when
    const result = solution(nums, n)

    // then
    assert.deepStrictEqual(-1, result)
    done();
  });

  it('test 6', function (done) {
    // given
    const nums = [1]
    const n = 1

    // when
    const result = solution(nums, n)

    // then
    assert.deepStrictEqual(0, result)
    done();
  });
})

File: 4기-코딩테스트-220608/4기 코딩테스트_220608/4기 - 1st 코딩테스트/SmallestSameIndexWithSorted/SmallestSameIndexWithSorted_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.medium.SmallestSameIndexWithSorted.SmallestSameIndexWithSorted import \
    solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        nums = [1, 3, 5, 3, 2]
        n = 3

        # when
        result = solution(nums, n)

        # then
        self.assertEqual(2, result)

    def test_2(self):
        # given
        usageArr = [2, 4, 2, 4, 2, 4, 4]
        fee = 4

        # when
        result = solution(usageArr, fee)

        # then
        self.assertEqual(3, result)

    def test_3(self):
        # given
        usageArr = [7, 7, 7, 6, 5, 10, 4, 4, 1, 8]
        fee = 6

        # when
        result = solution(usageArr, fee)

        # then
        self.assertEqual(4, result)

    def test_4(self):
        # given
        usageArr = []
        fee = 100

        # when
        result = solution(usageArr, fee)

        # then
        self.assertEqual(-1, result)

    def test_5(self):
        # given
        usageArr = [1, 3, 5, 3, 2]
        fee = 99

        # when
        result = solution(usageArr, fee)

        # then
        self.assertEqual(-1, result)

    def test_6(self):
        # given
        usageArr = [1]
        fee = 1

        # when
        result = solution(usageArr, fee)

        # then
        self.assertEqual(0, result)


if __name__ == '__main__':
    unittest.main()
